solidity/test/staker-rewards/MerkleDistributorTest.js

```diff
@@ -0,0 +1,210 @@
+const {contract, web3} = require("@openzeppelin/test-environment")
+const {expectRevert, expectEvent} = require("@openzeppelin/test-helpers")
+const {createSnapshot, restoreSnapshot} = require("../helpers/snapshot")
+const {testValues} = require("./rewardsData.js")
+
+const ECDSARewardsDistributor = contract.fromArtifact("ECDSARewardsDistributor")
+const KeepToken = contract.fromArtifact("KeepToken")
+
+const BN = web3.utils.BN
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("MerkleDistributor", () => {
+  let keepToken
+  let rewardsDistributor
+
+  before(async () => {
+    keepToken = await KeepToken.new()
+    rewardsDistributor = await ECDSARewardsDistributor.new(keepToken.address)
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("allocating rewards", () => {
+    it("should successfuly allocate KEEP tokens", async () => {
+      let balance = await keepToken.balanceOf(rewardsDistributor.address)
+      expect(balance).to.eq.BN(0)
+
+      await allocateTokens(
+        testValues.merkleObject0.merkleRoot,
+        testValues.merkleObject0.amountToAllocate
+      )
+
+      balance = await keepToken.balanceOf(rewardsDistributor.address)
+      expect(balance).to.eq.BN(testValues.merkleObject0.amountToAllocate)
+    })
+
+    it("should fail allocating KEEP tokens without prior approval", async () => {
+      await expectRevert(
+        rewardsDistributor.allocate(
+          testValues.merkleObject0.merkleRoot,
+          testValues.merkleObject0.amountToAllocate
+        ),
+        "SafeERC20: low-level call failed"
+      )
+    })
```

Comment:
Can we add another test ensuring `RewardsAllocated` event is emitted with correct values?

Reply:
Done.